t/13-joblocks.t

```diff
@@ -179,6 +182,31 @@ sub set_token_header {
         });
 }
 
+#######################################
+## Lock & unlock mutex on sibling jobs
+my $jP = job_create_with_worker('testA');
+my $jS1 = job_create_with_worker('testB', $jP);
+my $jS2 = job_create_with_worker('testC', $jP);
+
+# Create lock on sibling 1
+set_token_header($t->ua, 'token' . $jS1);
+$t->post_ok($m_prefix, form => {name => 'sblock'})->status_is(200);
+
+# Use lock from sibling 2
+set_token_header($t->ua, 'token' . $jS2);
+
+# Check it's locked in DB
+$t->post_ok($m_prefix . '/sblock', form => {action => 'lock', where => $jS1})->status_is(200);
+$res = $schema->resultset('JobLocks')->find({owner => $jS1, name => 'sblock'});
+is($res->locked_by, $jS2, 'mutex is locked');
+
+# Check it's unlocked in DB
+$t->post_ok($m_prefix . '/sblock', form => {action => 'unlock', where => $jS1})->status_is(200);
```

Comment:
this is just testing the happy path. What happens if you unlock first and then try to lock? what happens if you use a different name for unlock...

Reply:
Fixed, thanks